### ISSUE TITLE
Add a headings hierarchy checker notice in the heading block sidebar

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#10581]
Previously, the headings hierarchy was placed also in the _Document_ sidebar, in the "Table of Contents" panel:

![screen shot 2018-10-13 at 17 49 50](https://user-images.githubusercontent.com/1682452/46907382-d59c8b00-cf11-11e8-8813-a8b7b811fe37.png)

This panel has been removed so the only way to be aware of the incorrect headings level is to intentionally check the document outline tool in the top bar.

I'd like to propose to add a notice, even a small one, directly in the Heading block sidebar so users will be immediately informed the hierarchy is not correct _while_ they set the heading level. An option might be placing the notice under the buttons in the sidebar:

![screenshot 103](https://user-images.githubusercontent.com/1682452/46907423-7428ec00-cf12-11e8-93ed-743666eae872.png)

**Comment by mtias** [comment]
This is a great idea.

**Comment by afercia** [comment]
👍 A simple notice would be great and also consistent with the core patterns. Re: the wording, not sure if the message should actually suggest a fix telling "Please use..." but I have no strong opinions. For reference, see below the wording used in the "Content structure" popover:

<img width="380" alt="screenshot 2018-11-18 at 15 26 44" src="https://user-images.githubusercontent.com/1682452/48673937-29207b00-eb47-11e8-8392-223f4f09ff3f.png">

**Comment by getdave** [comment]
Thinking about this, one thing I've found quite common is to have the ability to visually style one  heading as though it were another heading level.

For example, say for some reason we need the 2nd level heading on the page to look like the `h1`. Currently, the only way to do this is by using the `h1` heading level/tag. However, I should be able to select `h2` and say "style this like a `h1`" and it should apply the `h1` style but leave the tag as a `h2`.

I know this sounds a bit odd, but in my experience, it's common that designs have headings that don't always visually follow the ideal hierarchical structure. This might help us avoid incorrect heading hierarchies.

**Comment by gziolo** [comment]
> 👍 A simple notice would be great and also consistent with the core patterns. Re: the wording, not sure if the message should actually suggest a fix telling "Please use..." but I have no strong opinions. For reference, see below the wording used in the "Content structure" popover:

We should align with the Color Settings panel and the notice that is shown when contrast verification fails:

![Screen Shot 2019-03-20 at 10 51 11](https://user-images.githubusercontent.com/699132/54675139-33a34f00-4afe-11e9-95fd-9cee6d658366.png)

When testing Color Settings I noticed that it is never announced for folks using VoiceOver so they might not be aware that there is an issue:
![Screen Shot 2019-03-20 at 10 55 08](https://user-images.githubusercontent.com/699132/54675370-b75d3b80-4afe-11e9-9942-a12809126bec.png)

Should we change it as well?

**Comment by afercia** [comment]
Yes! Ideally should work like all the other notices: when they appear, they should also be announced with `speak()`.

**Comment by mapk** [comment]
I really like @melchoyce's suggestion above. Let's get this developed @youknowriad.

**Comment by joedolson** [comment]
I like the suggestion from @melchoyce. The wording should probably be adjusted - in the example context, either an h2 or an h3 would be valid. I think that there are two paths we can take with the error message: either make it more generic, or more specific.

With any given heading, it can *always* be followed by a heading that's of the same or any higher level (excepting h1, as that should only be used once on any page) *or* by the heading immediate lower.

"higher" and "lower" levels of headings is ambiguous, however, which makes referencing these generically somewhat difficult. 

Making it specific will on rare occasion mean listing a large number of heading levels - but only in relatively rare cases, where somebody is actually using h4. (There are no invalid heading levels that can follow an h5 or an h6)

**Comment by gziolo** [comment]
@AmartyaU started working on it, the first PR opened for the missing voice announcement: #14649.

**Comment by karmatosed** [comment]
I am removing the 'User Experience (UX)' label as part of a label cleanup. It's not being used anymore consistently so let's try and keep to 'needs design' and 'needs design feedback'. If we find a need for another label we can consider it but having those 2 should cover this.

**Comment by skorasaurus** [comment]
also discussed in https://github.com/WordPress/gutenberg/issues/20870

**Comment by Mamaduka** [comment]
Closing as "maybe later". The document outline works, and block-level validations/lints need a more general solution instead of doing them on a per-block basis.

**Comment by afercia** [comment]
As I see it, this is an usability issue rather than a technical one.
I consider this issue still valid as the real problem here is the _discoverability_ of the feature and providing users with feedback _while_ they are adding headings.

As mentioned in the PR description:

> add a notice, even a small one, directly in the Heading block sidebar so users will be immediately informed the hierarchy is not correct _while_ they set the heading level


> block-level validations/lints need a more general solution

I'd like to repoen this issue. If there's no technical solution available yet, that's OK. But the underlying usability issue still needs to be solved. Thank you.
